src/scheduler/server_info.c

```diff
@@ -636,6 +636,12 @@ query_server_info(status *pol, struct batch_status *server)
 		attrp = attrp->next;
 	}
 
+	if (sinfo->job_sort_formula == NULL && sc_attrs.job_sort_formula != NULL) {
+		sinfo->job_sort_formula = string_dup(sc_attrs.job_sort_formula);
+		if (sinfo->job_sort_formula == NULL)
+			return NULL;
```

Comment:
If you return NULL, you'll leak the sinfo structure.  You need to free it first.